### PR TITLE
Added all the filters

### PR DESCRIPTION
**Pull request by anshikaa12**
>> added brands filter
>> added instock filter
>> added rating filter
>>added clear all filter

### CODE REVIEW

File: src/components/Card.jsx [hunk 1]
```diff
@@ -4,8 +4,13 @@ import { useProduct } from "../services/product_api";
 import { useWishlist } from "../context/wishlistcontext";
 import { useFilter } from "../context/filtercontext";
 import { sortedData } from "../context/filtercontext";
```
Comment (amarphule) [inline src/components/Card.jsx]: if both imports come from the same file you can use  `import {useFilter, sortedData } from "../context/filtercontext"`

File: src/components/Card.jsx [hunk 2]
```diff
@@ -14,16 +19,27 @@ function Card() {
   const { category } = useCategory();
   const { currState, wishFunc } = useWishlist();
   const { filterState, filterFunc } = useFilter();
+  console.log(filterState);
   const sortedData =
-    filterState.sortBy !== ""
+    filterState?.sortBy !== ""
```
Comment (amarphule) [inline src/components/Card.jsx]: why  ? after filterState

File: src/context/filtercontext.js [hunk 3]
```diff
@@ -1,19 +1,21 @@
 import { createContext, useContext, useReducer } from "react";
 import { filterReducer } from "../reducer/filterreducer";
-const filter = createContext();
-const useFilter = () => useContext(filter);
+const FilterContext = createContext();
 
 const FilterProvider = ({ children }) => {
   const [filterState, filterFunc] = useReducer(filterReducer, {
```
Comment (rehanmohiuddin) [inline src/context/filtercontext.js:6]: make constant of initialState and pass in useReducer it will be clean and scalable

**Review by rehanmohiuddin — APPROVED** [review]

**Review by amarphule — APPROVED** [review]
The code is well-formatted nice work. Given meaningful variable names, one suggestion use the camelCase format to follow a better practice

**Comment by anshikaa12** [comment]
> The code is well-formatted nice work. Given meaningful variable names, one suggestion use the camelCase format to follow a better practice

did the required changes!

**Pull request merged**